tmt/steps/execute/internal.py

```diff
@@ -34,7 +34,12 @@
 TEST_PIDFILE_LOCK_FILENAME = f'{TEST_PIDFILE_FILENAME}.lock'
 
 #: The default directory for storing test pid file.
-TEST_PIDFILE_ROOT = Path('/var/tmp')  # noqa: S108 insecure usage of temporary dir
+pid_directory = '/var/tmp/pid'  # noqa: S108 insecure usage of temporary dir
+Path(pid_directory).mkdir(parents=True, exist_ok=True)
+user_id = os.system("id -u")  # noqa: S605 Starting a process with a shell
+if user_id == os.stat(pid_directory).st_uid:
+    os.chmod(pid_directory, 0o777)  # noqa: S103 setting a permissive mask on file or directory
+TEST_PIDFILE_ROOT = Path(pid_directory)
```

Comment:
First, all this code needs to happen outside of the global, module-level scope - we do not want `tmt.steps.execute.internaluser_id` to exist, for example. We try to avoid import-time side effects, stuff happening when only importing tmt.

But there is a bigger problem, all this creates the directory on the machine running tmt, but that's the wrong place, it must exist (and have the right permissions) on the guest on which tests are running. Check the `setup()` method of `tmt.steps.provision.Guest` - this one runs to prepare the guest for testing, and I dare to say this would be the right place to extend, to create the directory on the guest.

Reply:
Pidfile is definitely created on the guest, see https://github.com/teemtee/tmt/blob/main/tmt/steps/execute/internal.py#L71. It is created by a shell wrapper `execute/internal` runs over SSH on the guest, and the wrapper creates the pidfile and then runs the test. It exists so scripts like `tmt-reboot` - which do run on the guest - could interrupt running test.

Reply:
@happz so what about adding something similar to:

```python
pid_directory = '/var/tmp/pid'
Path(pid_directory).mkdir(parents=True, exist_ok=True)
self.execute(ShellScript(f"chmod ugo+rwx {pid_directory}"))
```
At the end of the setup function?
https://github.com/teemtee/tmt/blob/f9642cf4374c9d2cea1334c777db37c0d6a6ecfa/tmt/steps/provision/__init__.py#L1689
Ps: I still see the warn message when trying with the second user :confused:

Reply:
Yes, I would try putting it into the `Guest.setup()` method.

Reply:
@happz with the following suggestion I still see the `pid` directory is created in the host running tmt

Reply:
Correct, because `Path(pid_directory).mkdir(parents=True, exist_ok=True)` is still running locally, on the machine running tmt. `self.execute(ShellScript(f"mkdir -p {pid_directory}"))` would translate this to the guest.

Reply:
Sorry for my obvious question... so now what happens with the TEST_PIDFILE_ROOT defined int the internal.py?

Reply:
Maybe I am misunderstanding something but that directory is being created in the TMT host

Reply:
Well, nothing happens to `TEST_PIDFILE_ROOT` :) It stays where it is, and what happens *to your patch* is that instead of having your own path in `pid_directory`, your code should use `tmt.steps.execute.internal.TEST_PIDFILE_ROOT` instead. That "constant" is correct, and should be used.